### PR TITLE
Fix #10153 fix wrong reset of details

### LINKED ISSUE #10153: Map Details are shown in new dashboard page
## Description

There is a problem in how details are managed for the map and dashboard,
the state is not perfectly cleared

## How to reproduce

- open a map with details
- go home
- create a new dashboard

*Expected Result*

details are missing

*Current Result*

still present in the sidebar

- [x] Not browser related

<details><summary> <b>Browser info</b> </summary>

(use this site: <a href="https://www.whatsmybrowser.org/">https://www.whatsmybrowser.org/</a> for non expert users)

| Browser Affected | Version |
|---|---|
|Internet Explorer| |
|Edge| |
|Chrome| |
|Firefox| |
|Safari| |
</details>

### PR DESCRIPTION
**Pull request by MV88**
## Description

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [x] Bugfix
 - [ ] Feature
 - [ ] Code style update (formatting, local variables)
 - [ ] Refactoring (no functional changes, no api changes)
 - [ ] Build related changes
 - [ ] CI related changes
 - [ ] Other... Please describe:

## Issue

**What is the current behavior?**

Fix #10153

**What is the new behavior?**

## Breaking change
**Does this PR introduce a breaking change?** (check one with "x", remove the other)
 - [ ] Yes, and I documented them in migration notes
 - [x] No

### CODE REVIEW

File: web/client/reducers/config.js [hunk 1]
```diff
@@ -98,7 +98,7 @@ function mapConfig(state = null, action) {
     case DETAILS_LOADED:
         let dashboardResource = state.dashboard?.resource;
         map = state && state.map && state.map.present ? state.map.present : state && state.map;
-        if (map && map.mapId.toString() === action.id.toString()) {
+        if (map && map?.mapId?.toString() === action.id.toString()) {
```
Comment (dsuren1) [inline web/client/reducers/config.js:101]:
```suggestion
        if (map && map.mapId?.toString() === action.id.toString()) {
```

File: web/client/reducers/config.js [hunk 2]
```diff
@@ -111,7 +111,7 @@ function mapConfig(state = null, action) {
                 }
             };
             return assign({}, state, {map: map});
-        } else if (dashboardResource && dashboardResource.id === action.id.toString()) {
+        } else if (dashboardResource && dashboardResource?.id?.toString() === action.id.toString()) {
```
Comment (dsuren1) [inline web/client/reducers/config.js:114]:
```suggestion
        } else if (dashboardResource && dashboardResource.id?.toString() === action.id.toString()) {
```

**Review by dsuren1 — CHANGES_REQUESTED** [review]

**Comment by dsuren1** [comment]
@ElenaGallo Kindly test this fix in dev and let us know if it's good to be backported. Thanks!

**Comment by ElenaGallo** [comment]
Test passed, @MV88 please backport to 2024.01.xx. Thanks

**Pull request merged**